Bio/codonalign/__init__.py

```diff
@@ -56,41 +54,30 @@ def build(
 
     The example below answers this Biostars question: https://www.biostars.org/p/89741/
 
-    >>> from Bio.Alphabet import generic_dna, generic_protein
     >>> from Bio.Seq import Seq
     >>> from Bio.SeqRecord import SeqRecord
     >>> from Bio.Align import MultipleSeqAlignment
     >>> from Bio.codonalign import build
-    >>> seq1 = SeqRecord(Seq('ATGTCTCGT', alphabet=generic_dna), id='pro1')
-    >>> seq2 = SeqRecord(Seq('ATGCGT', alphabet=generic_dna), id='pro2')
-    >>> pro1 = SeqRecord(Seq('MSR', alphabet=generic_protein), id='pro1')
-    >>> pro2 = SeqRecord(Seq('M-R', alphabet=generic_protein), id='pro2')
+    >>> seq1 = SeqRecord(Seq('ATGTCTCGT'), id='pro1')
+    >>> seq2 = SeqRecord(Seq('ATGCGT'), id='pro2')
+    >>> pro1 = SeqRecord(Seq('MSR'), id='pro1')
+    >>> pro2 = SeqRecord(Seq('M-R'), id='pro2')
     >>> aln = MultipleSeqAlignment([pro1, pro2])
     >>> codon_aln = build(aln, [seq1, seq2])
     >>> print(codon_aln)
-    CodonAlphabet(Standard) CodonAlignment with 2 rows and 9 columns (3 codons)
+    CodonAlignment with 2 rows and 9 columns (3 codons)
     ATGTCTCGT pro1
     ATG---CGT pro2
 
     """
     # TODO
     # add an option to allow the user to specify the returned object?
 
-    from Bio.Alphabet import ProteinAlphabet
     from Bio.Align import MultipleSeqAlignment
 
     # check the type of object of pro_align
     if not isinstance(pro_align, MultipleSeqAlignment):
         raise TypeError("the first argument should be a MultipleSeqAlignment object")
-    # check the alphabet of pro_align
-    for pro in pro_align:
-        if not isinstance(_get_base_alphabet(pro.seq.alphabet), ProteinAlphabet):
-            raise TypeError(
-                "Alphabet Error!\nThe input alignment should be "
-                "a *PROTEIN* alignment, found %r" % pro.seq.alphabet
-            )
-    if alphabet is None:
-        alphabet = _get_codon_alphabet(codon_table, gap_char=gap_char)
```

Comment:
It might be a good idea to check if the MSA is a protein alignment instead of nucleotide alignment here.

Reply:
The MSA won't store if it is a protein alignment or a nucleotide alignment.

Reply:
Well, it _could_ - but @mdehoon wasn't keen on my initial proposal on #3011.

Reply:
Okay. I think it may still be fine here. We have a check later to make sure the protein record and nucleotide record match.